src/runtime/task/mod.rs

```diff
@@ -343,6 +343,12 @@ impl Task {
         self.detached = true;
     }
 
+    pub(crate) fn abort(&mut self) {
+        self.finish();
+        let mut continuation = self.continuation.borrow_mut();
+        continuation.wipe();
```

Comment:
Maybe worth a comment about why `finish()` isn't good enough — we want to run the task's destructors now.

Reply:
Yes